pages/dashboard.js

```diff
@@ -45,39 +45,136 @@ export default function Dashboard(props) {
-  const [gisError, setGisError] = useState(false)
+  const [gisError, setGisError] = useState()
 
   const [sebBenefit, setSebBenefit] = useState()
   const [sebLoaded, setSebLoaded] = useState(false)
   const [sebError, setSebError] = useState()
 
+  const [allBenefits, setAllBenefits] = useState()
+
   useEffect(() => {
-    async function fetchProgramData(program, setData, setError, setLoaded) {
-      try {
-        const res = await fetch(`/api/programData/${program}`)
-        if (res.ok) {
-          if (res.status === 200) {
-            setData(await res.json())
+    if (
+      !cppLoaded &&
+      !cppdLoaded &&
+      !eiLoaded &&
+      !oasLoaded &&
+      !gisLoaded &&
+      !sebLoaded
+    ) {
+      async function fetchProgramData(program, setData, setError, setLoaded) {
+        try {
+          const res = await fetch(`/api/programData/${program}`)
+          if (res.ok) {
+            if (res.status === 200) {
+              setData(await res.json())
+            }
+          } else {
+            const text = await res.text()
+            setError(`Error fetching ${program} data ${res.status} - ${text}.`)
           }
-        } else {
-          const text = await res.text()
-          setError(`Error fetching ${program} data ${res.status} - ${text}.`)
+        } catch (error) {
+          setError(`Something went wrong fetching ${program} data.`)
+          console.log(error)
+        } finally {
+          setLoaded(true)
         }
-      } catch (error) {
-        setError(`Something went wrong fetching ${program} data.`)
-        console.log(error)
-      } finally {
-        setLoaded(true)
       }
+
+      fetchProgramData('cpp', setCppBenefit, setCppError, setCppLoaded)
+      fetchProgramData('cppd', setCppdBenefit, setCppdError, setCppdLoaded)
+      fetchProgramData('ei', setEiBenefit, setEiError, setEiLoaded)
+      fetchProgramData('oas', setOasBenefit, setOasError, setOasLoaded)
+      fetchProgramData('gis', setGisBenefit, setGisError, setGisLoaded)
+      fetchProgramData('seb', setSebBenefit, setSebError, setSebLoaded)
     }
+  }, [cppLoaded, cppdLoaded, eiLoaded, gisLoaded, oasLoaded, sebLoaded])
+
+  useEffect(() => {
+    if (
+      cppLoaded &&
+      cppdLoaded &&
+      eiLoaded &&
+      oasLoaded &&
+      gisLoaded &&
+      sebLoaded
+    ) {
+      const newArray = [
+        eiBenefit,
+        cppBenefit,
+        oasBenefit,
+        gisBenefit,
+        cppdBenefit,
+        sebBenefit,
+      ]
+
+      let orderedPaymentHold = []
+      let orderedApplicationReceived = []
+      let orderedDecisionSent = []
+      let orderedBenefitUpdate = []
+      let orderedInPayment = []
+      let orderedExhausted = []
+      let orderedInactive = []
+      let orderedActiveAgreement = []
+
+      newArray.forEach((benefits) => {
+        if (benefits != undefined) {
+          benefits.forEach((benefit) => {
+            switch (benefit.statusCode) {
```

Comment:
This is a really neat concept for sorting, I like, just wondering if you chose not to use `.sort(func)` for a particular reason?

Reply:
All good, using sort could be quite complex as it's all about evaluating the items within an array returning a positive or negative for each pair. As it's generally just used for numbers or alphabetical lists, using the custom override for something like this wouldn't be straight forward, what you've go here is easy to read.